### ISSUE TITLE
podman pull from docker.io - unauthorized

### ISSUE THREAD

**Issue by cybernet** [containers/podman#7780]
/kind bug

After a successful login with `podman login docker.io`

I cannot pull anything because authorization is required ...

**Steps to reproduce the issue:**

1. `podman login docker.io`

2. `podman pull pi-hole/pi-hole`

**Result**
```
[root@cent0 ~]# podman pull pi-hole/pi-hole
Trying to pull registry.access.redhat.com/pi-hole/pi-hole...
  name unknown: Repo not found
Trying to pull registry.redhat.io/pi-hole/pi-hole...
  unable to retrieve auth token: invalid username/password: unauthorized: Please login to the Red Hat Registry using your Customer Portal credentials. Further instructions can be found here: https://access.redhat.com/RegistryAuthentication
Trying to pull docker.io/pi-hole/pi-hole...
  denied: requested access to the resource is denied
Error: error pulling image "pi-hole/pi-hole": unable to pull pi-hole/pi-hole: 3 errors occurred:
	* Error initializing source docker://registry.access.redhat.com/pi-hole/pi-hole:latest: Error reading manifest latest in registry.access.redhat.com/pi-hole/pi-hole: name unknown: Repo not found
	* Error initializing source docker://registry.redhat.io/pi-hole/pi-hole:latest: unable to retrieve auth token: invalid username/password: unauthorized: Please login to the Red Hat Registry using your Customer Portal credentials. Further instructions can be found here: https://access.redhat.com/RegistryAuthentication
	* Error initializing source docker://pi-hole/pi-hole:latest: Error reading manifest latest in docker.io/pi-hole/pi-hole: errors:
denied: requested access to the resource is denied
unauthorized: authentication required

```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:            1.6.4
RemoteAPI Version:  1
Go Version:         go1.13.4
OS/Arch:            linux/amd64

```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  git commit: ""
  go version: go1.13.4
  podman version: 1.6.4
host:
  BuildahVersion: 1.12.0-dev
  CgroupVersion: v1
  Conmon:
    package: conmon-2.0.6-1.module_el8.2.0+305+5e198a41.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.6, commit: a2b11288060ebd7abd20e0b4eb1a834bbf0aec3e'
  Distribution:
    distribution: '"centos"'
    version: "8"
  MemFree: 3318145024
  MemTotal: 3960950784
  OCIRuntime:
    name: runc
    package: runc-1.0.0-65.rc10.module_el8.2.0+305+5e198a41.x86_64
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.1-dev'
  SwapFree: 4265603072
  SwapTotal: 4265603072
  arch: amd64
  cpus: 4
  eventlogger: journald
  hostname: cent0
  kernel: 4.18.0-193.19.1.el8_2.x86_64
  os: linux
  rootless: false
  uptime: 10m 27.61s
registries:
  blocked: null
  insecure: null
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  ConfigFile: /etc/containers/storage.conf
  ContainerStore:
    number: 0
  GraphDriverName: overlay
  GraphOptions: {}
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 3
  RunRoot: /var/run/containers/storage
  VolumePath: /var/lib/containers/storage/volumes

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-1.6.4-10.module_el8.2.0+305+5e198a41.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

No

**Comment by vrothberg** [comment]
Does it work if you pull by fully-qualified name `docker.io/pi-hole/pi-hole`?

**Comment by cybernet** [comment]
> Does it work if you pull by fully-qualified name `docker.io/pi-hole/pi-hole`?

no, it doesn't.
just curious I also tried with `docker://`

**Comment by vrothberg** [comment]
Did you login as root and pulled as root?

**Comment by rhatdan** [comment]
You must podman login first to get this image.

**Comment by vrothberg** [comment]
I'll re-open as the issue states that it's failing despite the login.

**Comment by rhatdan** [comment]
Does it succeed on Docker then, and Not on Podman?

**Comment by rhatdan** [comment]
Can you try with a newer version of Podman?

**Comment by cybernet** [comment]
> Can you try with a newer version of Podman?



> You must podman login first to get this image.

try reading again :smiley:

**Comment by cybernet** [comment]
> Did you login as root and pulled as root?

see output of the command - everything is done as root :)

**Comment by cybernet** [comment]
I have also tried with podman 2.1 from Solus distro, same result
Has any of you able to reproduce this ?

**Comment by vrothberg** [comment]
Can you try pulling another image? For instance, a private one form your own namespace?

Maybe you don't have permission to access pi-hole?

**Comment by rhatdan** [comment]
Could you attempt to pull the image with Skopeo and or buildah to see if they act differently?

**Comment by QiWang19** [comment]
This is the same issue as https://bugzilla.redhat.com/show_bug.cgi?id=1867495 that has been fixed. So close this.

**Comment by balthild** [comment]
Hello. The bugzilla link shows that the problem is fixed in podman v1.6.4, but I still encountered a similar problem in podman v3.0.1

```
[root@wsl ~]# podman login docker.io
Authenticating with existing credentials...
Existing credentials are valid. Already logged in to docker.io

[root@wsl ~]# podman pull php-7.3:fpm
Resolving "php-7.3" using unqualified-search registries (/etc/containers/registries.conf)
Trying to pull docker.io/library/php-7.3:fpm...
  denied: requested access to the resource is denied
Error: Error initializing source docker://php-7.3:fpm: Error reading manifest fpm in docker.io/library/php-7.3: errors:
denied: requested access to the resource is denied
unauthorized: authentication required

[root@wsl ~]# podman pull docker.io/php-7.3:fpm
Trying to pull docker.io/library/php-7.3:fpm...
  denied: requested access to the resource is denied
Error: Error initializing source docker://php-7.3:fpm: Error reading manifest fpm in docker.io/library/php-7.3: errors:
denied: requested access to the resource is denied
unauthorized: authentication required

[root@wsl ~]# podman version
Version:      3.0.1
API Version:  3.0.0
Go Version:   go1.16
Git Commit:   c640670e85c4aaaff92741691d6a854a90229d8d
Built:        Sun Feb 21 23:29:46 2021
OS/Arch:      linux/amd64
```

**Comment by cybernet** [comment]
> > Original comment
> > Edit: It's a typo... The correct image is `docker.io/php:7.3-fpm`. I'm sorry...
> 
> I get the image by `podman pull docker://uselagoon/php-7.3-fpm`

👍🏼
